src/components/vm/disks/diskAdd.jsx

```diff
@@ -182,25 +182,21 @@ const AdditionalOptions = ({ cacheMode, device, idPrefix, onValueChanged, busTyp
     // https://libvirt.org/formatdomain.html#hard-drives-floppy-disks-cdroms
     const serialLength = busType === "scsi" ? 36 : 20;
 
-    const setSerialHelper = useCallback(value => {
-        const clearedSerial = clearSerial(value);
+    useEffect(() => {
+        const clearedSerial = clearSerial(serial);
 
-        if (value !== clearedSerial)
+        if (serial !== clearedSerial) {
             // Show the message once triggerred and leave it around as reminder
             setShowAllowedCharactersMessage(true);
+            onValueChanged('serial', clearedSerial);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1303-20231109-142627-1ac1b404-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)